### PR TITLE
hdf5: add develop-2.0

### PR DESCRIPTION
**Pull request by lrknox**

### CODE REVIEW

**Review by hyoklee — APPROVED** [review]
Thanks for the quick patch, @lrknox !

**Review by bendamasio — COMMENTED** [review]
Hello @brtnfld, @byrnHDF, @gheber, @hyoklee and @lkurz, I generated some build results:<br>
:information_source: `spack install` on the changed recipes of this PR was successful[^1] on Ubuntu 22.04.4 LTS!<details><summary>Click here for a summary of the successful builds.</summary>
These changed specs were found using [gh pr diff](https://cli.github.com):
- `hdf5@develop-2.0`
- `hdf5@develop-2.0`
- `hdf5@develop-1.14`
- `hdf5@develop-1.12`


<details><summary><b>Click here to show/hide the report on </b><tt>spack find</tt> for <tt>linux-ubuntu22.04-skylake / gcc@13.1.0</tt>.</summary><ol>

<li><details><summary>Click on this area to unfold/refold the dependency report of this build:

>```py
>hdf5@develop-1.12~cxx~fortran~hl~ipo~java~map+mpi+shared~subfiling~szip~threadsafe+tools api=default build_system=cmake build_type=Release generator=make
>```
</summary>

>```py
>	cmake@3.30.5
>		curl@8.10.1 tls=openssl
>			nghttp2@1.63.0
>				diffutils@3.10
>			openssl@3.3.1 certs=mozilla
>				ca-certificates-mozilla@2023-05-30
>		ncurses@6.5
>	gcc-runtime@13.1.0
>	glibc@2.35
>	gmake@4.4.1
>	openmpi@5.0.5atomics-hwloc-libevent-pmix-rpath
>		autoconf@2.72
>			m4@1.4.19sigsegv,bfdffa7
>				libsigsegv@2.14
>		automake@1.16.5
>		hwloc@2.11.1-level-zero
>			libpciaccess@0.17
>				util-macros@1.20.1
>			libxml2@2.10.3
>				xz@5.4.6
>		libevent@2.1.12openssl
>		libtool@2.4.7
>			findutils@4.9.0
>		numactl@2.0.18
>		openssh@9.8p1gssapi
>			krb5@1.21.3
>				bison@3.8.2
>				gettext@0.22.5bzip2cursesgit
>					tar@1.34 zip=pigz
>						pigz@2.8
>						zstd@1.5.6programs
>			libedit@3.1-20240808
>			libxcrypt@4.4.35_api
>		perl@5.40.0cpanmopcodeopenthreads
>			berkeley-db@18.1.40cxx,b231fcc
>			bzip2@1.0.8
>			gdbm@1.23
>				readline@8.2
>		pkgconf@2.2.0
>		pmix@5.0.3
>	pkgconf@2.2.0
>		gmake@4.4.1
>	zlib-ng@2.2.1compat_strategiesopt
>
>```

</details>


<li><details><summary>Click on this area to unfold/refold the dependency report of this build:

>```py
>hdf5@develop-1.14~cxx~fortran~hl~ipo~java~map+mpi+shared~subfiling~szip~threadsafe+tools api=default build_system=cmake build_type=Release generator=make
>```
</summary>

>```py
>	cmake@3.30.5
>		curl@8.10.1 tls=openssl
>			nghttp2@1.63.0
>				diffutils@3.10
>			openssl@3.3.1 certs=mozilla
>				ca-certificates-mozilla@2023-05-30
>		ncurses@6.5
>	gcc-runtime@13.1.0
>	glibc@2.35
>	gmake@4.4.1
>	openmpi@5.0.5atomics-hwloc-libevent-pmix-rpath
>		autoconf@2.72
>			m4@1.4.19sigsegv,bfdffa7
>				libsigsegv@2.14
>		automake@1.16.5
>		hwloc@2.11.1-level-zero
>			libpciaccess@0.17
>				util-macros@1.20.1
>			libxml2@2.10.3
>				xz@5.4.6
>		libevent@2.1.12openssl
>		libtool@2.4.7
>			findutils@4.9.0
>		numactl@2.0.18
>		openssh@9.8p1gssapi
>			krb5@1.21.3
>				bison@3.8.2
>				gettext@0.22.5bzip2cursesgit
>					tar@1.34 zip=pigz
>						pigz@2.8
>						zstd@1.5.6programs
>			libedit@3.1-20240808
>			libxcrypt@4.4.35_api
>		perl@5.40.0cpanmopcodeopenthreads
>			berkeley-db@18.1.40cxx,b231fcc
>			bzip2@1.0.8
>			gdbm@1.23
>				readline@8.2
>		pkgconf@2.2.0
>		pmix@5.0.3
>	pkgconf@2.2.0
>		gmake@4.4.1
>	zlib-ng@2.2.1compatnew_strategiesopt
>
>```

</details>


<li><details><summary>Click on this area to unfold/refold the dependency report of this build:

>```py
>hdf5@develop-2.0~cxx~fortran~hl~ipo~java~map+mpi+shared~subfiling~szip~threadsafe+tools api=default build_system=cmake build_type=Release generator=make
>```
</summary>

>```py
>	cmake@3.30.5
>		curl@8.10.1 tls=openssl
>			nghttp2@1.63.0
>				diffutils@3.10
>			openssl@3.3.1 certs=mozilla
>				ca-certificates-mozilla@2023-05-30
>		ncurses@6.5
>	gcc-runtime@13.1.0
>	glibc@2.35
>	gmake@4.4.1
>	openmpi@5.0.5atomics-hwloc-libevent-pmix-rpath
>		autoconf@2.72
>			m4@1.4.19sigsegv,bfdffa7
>				libsigsegv@2.14
>		automake@1.16.5
>		hwloc@2.11.1-level-zero
>			libpciaccess@0.17
>				util-macros@1.20.1
>			libxml2@2.10.3
>				xz@5.4.6
>		libevent@2.1.12openssl
>		libtool@2.4.7
>			findutils@4.9.0
>		numactl@2.0.18
>		openssh@9.8p1gssapi
>			krb5@1.21.3
>				bison@3.8.2
>				gettext@0.22.5bzip2cursesgit
>					tar@1.34 zip=pigz
>						pigz@2.8
>						zstd@1.5.6programs
>			libedit@3.1-20240808
>			libxcrypt@4.4.35_api
>		perl@5.40.0cpanmopcodeopenthreads
>			berkeley-db@18.1.40cxx,b231fcc
>			bzip2@1.0.8
>			gdbm@1.23
>				readline@8.2
>		pkgconf@2.2.0
>		pmix@5.0.3
>	pkgconf@2.2.0
>		gmake@4.4.1
>	zlib-ng@2.2.1compatnew_strategiesopt
>
>```

</details>



</details>

[^1]: Orchestrated using `spack install` by [github.com/spack/gh-spack-pr](https://github.com/spack/gh-spack-pr) and the [GitHub CLI](https://cli.github.com/)
</details><details><summary>Click here for a disclaimer for the maintainers of the recipes

</summary><br>👍 Hello @brtnfld, @byrnHDF, @gheber and @lkurz!<br><br>

- We would like to ask you if you could review this PR within 5 days.
- If the PR is not reviewed within 5 days, the PR can be approved and merged by members.
- Please use 👍, Approve, LGTM or Request the comment so we can act accordingly.
- If you like to review but need more time, please add a comment to the PR.
- If you have questions (or would like to be removed as maintainer), please let us know.

Thank you for your time and effort!<br>
You can also join the Spack Slack channel for more information and have a chat:
https://spackpm.slack.com → #pull-requests. Thank you and best regards, @x64</details>

**Review by byrnHDF — APPROVED** [review]

**Review by bernhardkaindl — APPROVED** [review]

**Review by bernhardkaindl — APPROVED** [review]
While we have maintainer approval, I think it's best to leave other maintainers or spack members some time to comment. Or another more competent maintainer can merge it.

If needed quick, ping me (@bernhardkaindl) or another maintainer to merge.

**Review by alalazo — APPROVED** [review]

**Pull request merged**
